Plugins/GitUIPluginInterfaces/Extensibility/ISettingsSource.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace GitUIPluginInterfaces.Extensibility
+{
+    public interface ISettingsSource
+    {
+        T GetValue<T>(string key, T defaultValue = default!);
```

Comment:
The `!` here seems like a problem. Can you experiment with some different options here?

What will the implementation return if `defaultValue` is `null`? 

It's possible to annotate this as (from memory):

```suggestion
        [return: NotNullIfNotNull("defaultValue")]
        T? GetValue<T>(string key, T? defaultValue = default);
```

`GetValue` and `SetValue` should be documented.

Reply:
I think this operator for this case.

`var person = new Person(null!);`

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-forgiving

> What will the implementation return if defaultValue is null?

In runtime will be ok. With null as value will be ok.

Reply:
> GetValue and SetValue should be documented.

👍

Reply:
I'm familiar with the `!` operator, and we should avoid using it as it can hide bugs. The Nullable Reference Types feature is design-time only, so it's true that this code might be fine, but consuming code will assume the value cannot be null, and then may receive a null at runtime and blow up.

I think the annotations here have to be changed.

Reply:
👍